Automattic/jetpack
# CLI: add optional analytics tracking

## Proposed changes:

- Adds optional analytics tracking in Jetpack CLI using Tracks.
- Adds `jetpack cli analytics <on|off>` to set analytics tracking preference.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/garage-backlog/issues/57

## Does this pull request change what data or activity we track or use?

Yes, this PR adds optional Tracks data collection to Jetpack CLI. The purpose of this collection is to provide general metrics to inform on Jetpack CLI development projects.

## Testing instructions:

- Pull down changes and run `jetpack rsync --watch` which should prompt you to enable analytics (for testing purposes say yes). This should only ask you once.
- You can bail out of the sync process, we just wanted to fire the `tracks()` function.
- After some time (5-10 minutes), you should see a `jetpack_cli_rsync_watch` record over at the Tracks Live View page.
- The recorded event should have a `uuid` in the `eventprops` which matches the one stored in `~/.config/configstore/automattic/jetpack-cli/cli.json`
- Try running `jetpack cli analytics` which should ask you to specify the 'on' or 'off' positional. Disable for now.
- If analytics are disabled, then `jetpack rsync --watch` should not fire the Tracks event.

## Todos:

- [x] Remove 'test' prefix from event name.
- [x] After merge, formally validate and register the new Tracks event. [ Done in https://github.com/Automattic/tracks-events-registration/pull/2328 ]
- [x] After merge, give notice to devs. [ Done in pdWQjU-Ks-p2 ]
- [x] Ensure any privacy considerations have been reviewed.
